docs/dev/metrics.md

```diff
@@ -0,0 +1,108 @@
+
+6. **Avoid Redundancy in Naming**: Avoid repeating terms that are implied by the context or domain, making metric names shorter and more readable. Example: `results_cache_hits` instead of `results_cache_cache_hits`, `query_duration_ms` instead of `query_query_duration_ms`.
+   - **Why**: Redundancy adds unnecessary complexity and reduces readability. Shorter, more focused names enhance understanding and ease of use (Developer Experience principle).
+
+7. **Counter Naming Conventions**: Avoid `_total` suffix in counter names e.g., `flight_requests` instead of `flight_requests_total`.   - **Why**: Standardized counter naming aligns with Prometheus conventions, reducing confusion and supporting Developer Experience by avoiding unnecessary naming variation.
+
+8. **UpDownCounter Naming**: Use `_count` suffix and pluralization only when recording discrete instances of countable items, e.g., `db_client_active_connections_count`.
+   - **Why**: Clearly distinguishing counter-based metrics improves interpretability, which enhances the Developer Experience by making data analysis more intuitive.
+
+9. **Include Units in Name for Unit-Based Metrics**: Specify the unit (e.g., `_ms` for milliseconds, `_bytes` for data size) within metric names where applicable. Units should adhere to UCUM standards for consistent scientific notation.
+   - **Why**: Including UCUM-adherent units in names prevents confusion over metric scale, especially when integrating with external tools, and reinforces Clarity and Developer Experience by providing universally understood units.
+
+10. **Use Default Units for Common Metrics**:
+   - **Family**: **Unit**
+   - Timestamps and Duration: `ms`
+    - Data size: `bytes`
+   - **Why**: Establishing default units ensures predictability, reduces potential scaling errors, and simplifies interpretation, aligning with Developer Experience by maintaining consistency.
+
+## References and Standards Alignment
+
+This naming proposal integrates OpenTelemetry, Prometheus, and UCUM standards, combining Prometheus’s extensive adoption in metrics collection with OpenTelemetry’s semantic conventions and UCUM’s scientific accuracy for unit naming.
+
+- [Prometheus Naming Best Practices](https://prometheus.io/docs/practices/naming/)
+- [OpenTelemetry Metric Semantic Conventions](https://opentelemetry.io/docs/specs/semconv/general/metrics/)
+- [OpenTelemetry Attribute Naming](https://opentelemetry.io/docs/specs/semconv/general/attribute-naming/)
+- [UCUM Standard for Units](https://ucum.org/trac)
+
+### Notable Differences
+
+- **From OpenTelemetry**:
+  - Uses underscores (`_`) for domain separation instead of dots.
+  - Explicitly includes unit suffixes (e.g., `query_execution_duration_ms` instead of `query.operation.duration`).
+- **From Prometheus**:
+  - Does not use `_total` suffix for counters.
+  - Default units favor milliseconds (`ms`) over Prometheus’s preferred base unit of seconds.
```

Comment:
Add we use `_bytes` instead of `_by`

Reply:
Just to confirm: this is to add for OTEL difference as Prometheus is using `bytes`? Going to add this:
```
- Uses the `_bytes` unit suffix instead of the UCUM-recommended `By` unit
```

Reply:
I think its fine to leave out with the above note